Lombiq.Tests.UI.Samples/Tests/InteractiveModeTests.cs

```diff
@@ -34,57 +31,19 @@ public Task SampleTest() =>
                 await context.SignInDirectlyAndGoToDashboardAsync();
-                await context.SwitchToInteractiveAsync();
+                await context.SwitchToInteractiveAsync(cancellationToken: context.Configuration.TestCancellationToken);
 
                 // Afterwards if you can still evaluate code as normal so `SignInDirectlyAndGoToDashboardAsync()` can be
                 // inserted anywhere. Bare in mind, that it's safest to use it before code that's already going to
                 // navigate away, like `GetCurrentUserNameAsync()` here. This ensures any manual navigation you did
                 // won't affect the test. If that's not practical, you can also do your thing in a new tab and close it
                 // before continuing the test.
                 (await context.GetCurrentUserNameAsync()).ShouldNotBeNullOrWhiteSpace();
-            });
 
-    // This test checks if interactive mode works by opening it in one thread, and then clicking it away in a different
-    // thread. Two threads are necessary because interactive mode stops test execution on its current thread, so we
-    // wouldn't be able to end it from within a test.
-    [Fact]
-    public Task EnteringInteractiveModeShouldWait() =>
-        ExecuteTestAfterSetupAsync(
-            async context =>
-            {
-                var currentUrl = context.Driver.Url;
-
-                await Task.WhenAll(
-                    context.SwitchToInteractiveAsync(),
-                    Task.Run(
-                        async () =>
-                        {
-                            try
-                            {
-                                ReliabilityHelper.DoWithRetriesOrFail(
-                                    () => context.Driver.WindowHandles.Count > 1,
-                                    TimeSpan.FromSeconds(15));
-
-                                context.SwitchToLastWindow();
-
-                                await context.ClickReliablyOnAsync(By.ClassName("interactive__continue"));
-                            }
-                            catch (Exception ex)
-                            {
-                                _testOutputHelper.WriteLineTimestampedAndDebug(
-                                    "Interactive mode wasn't canceled properly due to the following exception. Canceling the test. {0}",
-                                    ex);
-
-                                // The other thread will wait indefinitely if the button wasn't clicked in the end. So,
-                                // failing the test then.
-                                TestContext.Current.CancelCurrentTest();
-                            }
-                        },
-                        context.Configuration.TestCancellationToken));
-
-                // Ensure that the info tab is closed and the control handed back to the last tab.
-                context.Driver.Url.ShouldBe(currentUrl);
-            });
```

Comment:
I've spent half a day trying to make this reliable, i.e. not hang randomly (that hung the tests too), and be reliable with its interactions. I couldn't get there, it was always flaky. I don't see any point in trying harder, we don't really need this test.